cdap-api/src/main/java/co/cask/cdap/api/lineage/LineageRecorder.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import co.cask.cdap.api.lineage.operation.Operation;
+
+import java.util.Collection;
+
+/**
+ * Interface for recording the lineage information.
+ */
+public interface LineageRecorder {
+  /**
+   * Record the lineage operations.
+   * @param operations the collection of operations to be recorded.
```

Comment:
I think it needs to be pointed out that the names of operations must be unique. And in addition all other requirements.

Reply:
updated the doc.